x-pack/plugins/security_solution/public/common/hooks/endpoint/use_alert_response_actions_support.ts

```diff
@@ -0,0 +1,241 @@
+        { category: 'observer', field: RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD.sentinel_one },
+        eventData
+      );
+    }
+
+    if (agentType === 'crowdstrike') {
+      return getAlertDetailsFieldValue(
+        { category: 'crowdstrike', field: RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD.crowdstrike },
+        eventData
+      );
+    }
+
+    return '';
+  }, [agentType, eventData]);
+
+  const doesHostSupportResponseActions = useMemo(() => {
+    return Boolean(isFeatureEnabled && isAlert && agentId && agentType);
+  }, [agentId, agentType, isAlert, isFeatureEnabled]);
+
+  const agentIdField = useMemo(() => {
+    if (agentType) {
+      return RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD[agentType];
+    }
+
+    return '';
+  }, [agentType]);
+
+  const supportedActions = useMemo(() => {
+    return RESPONSE_ACTION_API_COMMANDS_NAMES.reduce<AlertAgentActionsSupported>(
+      (acc, responseActionName) => {
+        acc[responseActionName] = false;
+
+        if (agentType && isFeatureEnabled) {
+          acc[responseActionName] = isAgentTypeAndActionSupported(
+            agentType,
+            responseActionName,
+            'manual'
+          );
+        }
+
+        return acc;
+      },
+      {} as AlertAgentActionsSupported
+    );
+  }, [agentType, isFeatureEnabled]);
+
+  const hostName = useMemo(() => {
+    // TODO:PT need to check if crowdstrike event has `host.name`
+    if (agentType === 'crowdstrike') {
+      return getAlertDetailsFieldValue(
+        { category: 'crowdstrike', field: 'crowdstrike.event.HostName' },
+        eventData
+      );
+    }
+
+    return getAlertDetailsFieldValue({ category: 'host', field: 'host.name' }, eventData);
+  }, [agentType, eventData]);
+
+  const platform = useMemo(() => {
+    // TODO:PT need to check if crowdstrike event has `host.os.family`
+    if (agentType === 'crowdstrike') {
+      return getAlertDetailsFieldValue(
+        { category: 'crowdstrike', field: 'crowdstrike.event.Platform' },
+        eventData
+      );
+    }
+
+    return getAlertDetailsFieldValue({ category: 'host', field: 'host.os.family' }, eventData);
+  }, [agentType, eventData]);
+
+  const unsupportedReason = useMemo(() => {
+    if (!doesHostSupportResponseActions) {
+      if (!isAlert) {
+        return RESPONSE_ACTIONS_ONLY_SUPPORTED_ON_ALERTS;
+      }
+
+      if (!agentType) {
+        // No message is provided for this condition because the
+        // return from this hook will always default to `endpoint`
+        return;
+      }
+
+      if (!agentId) {
+        return ALERT_EVENT_DATA_MISSING_AGENT_ID_FIELD(getAgentTypeName(agentType), agentIdField);
+      }
+    }
+  }, [agentId, agentIdField, agentType, doesHostSupportResponseActions, isAlert]);
+
+  return useMemo<AlertResponseActionsSupport>(() => {
+    return {
+      isSupported: doesHostSupportResponseActions,
+      unsupportedReason,
+      isAlert,
+      details: {
+        agentType: agentType || 'endpoint',
```

Comment:
I don't think this is true: "At this point we should definitely have an agentType"

One of the main purposes of this new hook is to discourage us to doing "agentType" checks outside of this hook. You basically call it with an Event Detail data and this hook will return back a data structure that tells you if response actions are supported or not and if supported, then many of these `details` will be filled in.

Also,
I did not want to make `agentType` here accept `undefined` because then we would have to cast or do type guard assertion when trying to use it, thus why it defaults to `endpoint`

Reply:
I'm actually concerned with mismatch between `details.agentType` and other values in the returned value. For example, platform doesn't depend on agentType for _non-crowdstrike_ agents (including `undefined`) and I'm wondering if there would be a case that this info is different when we force the agentType from `undefined` to `endpoint`.

Reply:
The thing is that the data inside of `details` should only be used if the outer property `isSupported` is set to `true`. And that property will always be `false` if we can't determine the `agentType`. 

I'm happy to re-look at this and maybe make the entire `details` be `undefined` if `isSupported` is `false`... but I'm not seeing the need right now. If you have an example of an alert that could cause an issue, let me know.